### PR TITLE
Added swap to VecModel

### LINKED ISSUE #5963: Add VecModel::swap
Another needed method is [`swap()`](https://doc.rust-lang.org/std/primitive.slice.html#method.swap). I needed this for up- and down-buttons with `StandardListView`s to move items.

_Originally posted by @Enyium in https://github.com/slint-ui/slint/issues/5881#issuecomment-2308354804_

### PR DESCRIPTION
**Pull request by ghost**
Fixes #5963

### CODE REVIEW

**Review by ogoffart — APPROVED** [review]

File: internal/core/model.rs [hunk 1]
```diff
@@ -376,6 +376,13 @@ impl<T: 'static> VecModel<T> {
         self.array.borrow_mut().clear();
         self.notify.reset();
     }
+
+    /// Swaps two elements in the model.
+    pub fn swap(&self, a: usize, b: usize) {
+        self.array.borrow_mut().swap(a, b);
+        self.notify.row_changed(a);
+        self.notify.row_changed(b);
```
Comment (tronical) [inline internal/core/model.rs:384]: Shouldn't this avoid a notification if `a == b`? (and be tested?)

**Comment by ogoffart** [comment]
true, i forgot about tests. (Same for VecModel::clear)

**Comment by tronical** [comment]
Beyond the scope of this PR, the model tracking could also have swapping as a function on the trait (that defaults to two row_changed calls). That would allow optimising the swap in some of the model adapters.

**Comment by ogoffart** [comment]
> Beyond the scope of this PR, the model tracking could also have swapping as a function on the trait (that defaults to two row_changed calls). That would allow optimising the swap in some of the model adapters.

Qt has a thing with [changePersistentIndexList](https://doc.qt.io/qt-6/qabstractitemmodel.html#changePersistentIndexList) to be able to do any mapping or sort or whatever. 
Not sure how best do that for the model tracker. Maybe could be done with a `fn map(&dyn Mapper)` on the ModelTracker and ModelNotify but that's outside of the scope of this PR anyway.

**Pull request merged**